src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -152,17 +152,17 @@ private X509Certificate ServerCertSelectionCallbackWrapper(string targetHost)
 
         private SslAuthenticationOptions CreateAuthenticationOptions(SslServerAuthenticationOptions sslServerAuthenticationOptions)
         {
-            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null)
+            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null && _certSelectionDelegate == null)
             {
                 throw new ArgumentNullException(nameof(sslServerAuthenticationOptions.ServerCertificate));
             }
 
-            if (sslServerAuthenticationOptions.ServerCertificate != null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback != null)
+            if (sslServerAuthenticationOptions.ServerCertificate != null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback != null  && _certSelectionDelegate == null)
```

Comment:
what if all 3 are not null?

Reply:
That's actually necessarily the case when `LocalCertificateSelectionCallback` is set -- if my understanding is correct we still need to provide a certificate to `AuthenticateAsServerAsync`, even though we will pick the actual certificate using the callback. That's actually the case tested in the first test I added.

Reply:
IMO we need to leave the line as it was before. This particular check should not be affected by the fact `_certSelectionDelegate` is set or not.

Reply:
Leaving this line as it was before would mean throwing an exception when both methods are used instead of defaulting to the new one. That isn't an unreasonable behavior, but it isn't what we originally discussed.

Reply:
I've removed my change in the latest commit -- this line now ensures that we throw an exception when both callbacks are set.